collector/lib/SysdigService.cpp

```diff
@@ -60,7 +69,9 @@ void SysdigService::Init(const CollectorConfig& config, std::shared_ptr<Connecti
     AddSignalHandler(MakeUnique<ProcessSignalHandler>(inspector_.get(), config.grpc_channel, &userspace_stats_));
   }
 
-  if (signal_handlers_.empty()) {
+  if (!conn_tracker && !config.grpc_channel) {
```

Comment:
It is because of the two if statements above. We only add other handlers when `conn_tracker == true` and `config.grpc_channel != nullptr` so if both of those are false (or rather `false && nullptr`) then the signal handlers list is empty except for our self check handlers